src/engine/input/key_mapping/SpecifiedInputKey.cs

```diff
@@ -59,7 +59,23 @@ public override string ToString()
         {
             // TODO: add translation support for key names like Right, and Left etc.
             // TODO: https://github.com/Revolutionary-Games/Thrive/issues/1895
-            text += ((KeyList)Code).ToString();
+
+            // text += ((KeyList)Code).ToString();
+
+            var keyname = ((KeyList)Code).ToString();
+
+            if (keyname == "Left")
+                text += TranslationServer.Translate("LEFT");
+            else if (keyname == "Right")
+                text += TranslationServer.Translate("RIGHT");
+            else if (keyname == "Up")
+                text += TranslationServer.Translate("UP");
+            else if (keyname == "Down")
+                text += TranslationServer.Translate("DOWN");
+            else if (keyname == "Print")
+                text += TranslationServer.Translate("PRINT");
+            else
+                text += ((KeyList)Code).ToString();
```

Comment:
due to KeyList being an enum I would not call the ToString and compare the string, I would compare the enum values.
```suggestion
            var key = (KeyList)Code;

            text += key switch
            {
                KeyList.Left => TranslationServer.Translate("LEFT"),
                KeyList.Right => TranslationServer.Translate("RIGHT"),
                KeyList.Up => TranslationServer.Translate("UP"),
                KeyList.Down => TranslationServer.Translate("DOWN"),
                KeyList.Print => TranslationServer.Translate("PRINT"),
                _ => key.ToString(),
            };
```